### PR TITLE
Fix unspent inputs

### PR DESCRIPTION
**Pull request by escottalexander**
This updates pending inputs of invalid transactions back to an unspent status.

### CODE REVIEW

File: packages/bitcore-node/migrations/fixUnspentInputs.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            if (!Storage.connected) {
+                await Storage.start(storageArgs);
+                await wait(2000);
+              }
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    async runScript() {
```
Comment (kajoseph) [inline packages/bitcore-node/migrations/fixUnspentInputs.js:39]: Let's add a param to do a dry-run that outputs the txids that will be changed. By default, it should do a dry-run and then have a cli param to run it for real

File: packages/bitcore-node/migrations/fixUnspentInputs.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/****************************************
```
Comment (kajoseph) [inline packages/bitcore-node/migrations/fixUnspentInputs.js:1]: Add a shebang to the top of the file `#!/usr/bin/env node` and give this file execution permissions `chmod +x fixUnspentInputs.js`

**Review by kajoseph — CHANGES_REQUESTED** [review]
I think you're on the right track - thanks for doing this. Just some relatively minor things and a few best-practice items

File: packages/bitcore-node/migrations/fixUnspentInputs.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    async runScript() {
+        console.log('running script');
+            for await (const chain of RBFEnabledUTXOChains) {
+                for await (const network of networks) {
+                    // using Set so that we don't have duplicates
+                    console.log(`updating records for ${chain}:${network}`);
+                    // Get array of all invalid txs for chain and network
+                    const invalidTxs = await this.transactionModel.collection.find({ chain, network, blockHeight: -3 }).toArray();
```
[the remark this answers is not in the document]
Comment (kajoseph) [reply packages/bitcore-node/migrations/fixUnspentInputs.js:46]: Ah ok. Might be good to just make a local const then

**Review by mohammadfarari1360 — COMMENTED** [review]
Okay

**Comment by escottalexander** [comment]
> I think you're on the right track - thanks for doing this. Just some relatively minor things and a few best-practice items

Thanks for your input. I made the revisions you suggested. Let me know if you see anything else.

**Review by kajoseph — APPROVED** [review]
Looks good

**Review by mohammadfarari1360 — COMMENTED** [review]
Okay good

**Pull request merged**